### PR TITLE
'stdin' is not a valid file extension for bash completions

### LINKED ISSUE #4803: "stdin" file extensions appear in autocomplete
https://github.com/kubernetes/kubernetes/pull/13482 refactored to use the same list for printing help, generating bash autocomplete, and filtering file extensions for directory visitors.

As a result, "stdin" appears in the bash file extension autocomplete, which doesn't make sense.

Accepted file extensions and whether stdin is accepted should be handled independently

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes #4803

"stdin" should only be used in help and messages, like when running `oc create` without a filename. Autocomplete and directory visitors only take actual file extensions (json, yaml, yml).

```
$oc create 
error: you must provide one or more resources by argument or filename (.json|.yaml|.yml|stdin)
```

@liggitt if you like the strategy I'll propose it upstream. Notice this also removes the dot before 'stdin' when printing help.

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -32,7 +32,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-var FileExtensions = []string{".json", ".stdin", ".yaml", ".yml"}
+var FileExtensions = []string{".json", ".yaml", ".yml"}
+var InputExtensions = append(FileExtensions, "stdin")
```
Comment (fabianofranz) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go]: @liggitt better name? `InputSources`?

**Comment by liggitt** [comment]
seems fine. remember to update the commit with the upstream PR number

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**